include/boost/math/differentiation/autodiff.hpp

```diff
@@ -567,13 +566,36 @@ template<typename RealType, size_t Order, size_t... Orders>
 using autodiff_fvar = typename detail::nest_fvar<RealType,Order,Orders...>::type;
 
 template<typename RealType, size_t Order, size_t... Orders>
-autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca)
-{
+autodiff_fvar<RealType,Order,Orders...> make_fvar(const RealType& ca) {
   return autodiff_fvar<RealType,Order,Orders...>(ca, true);
 }
 
 namespace detail {
 
+template <size_t>
+struct zero : std::integral_constant<size_t,0> {};
+
+template<typename RealType, size_t Order, size_t... Is>
+auto make_fvar_for_tuple(std::index_sequence<Is...>, const RealType& ca) {
+  return make_fvar<RealType,zero<Is>::value...,Order>(ca);
+}
+
+template<typename RealType, size_t... Orders, size_t... Is, typename... RealTypes>
+auto make_ftuple(std::index_sequence<Is...>, const RealTypes&... ca) {
+  return std::make_tuple(make_fvar_for_tuple<RealType,Orders>(std::make_index_sequence<Is>{},ca)...);
+}
+
+} // namespace detail
+
+template<typename RealType, size_t... Orders, typename... RealTypes>
+auto make_ftuple(const RealTypes&... ca) {
+  static_assert(sizeof...(Orders) == sizeof...(RealTypes),
+    "Number of Orders must match number of function parameters.");
+  return detail::make_ftuple<RealType,Orders...>(std::index_sequence_for<RealTypes...>{}, ca...);
+}
```

Comment:
I'm having some trouble correlating this with the example code but I also note the build failed so maybe this is out of date?

Reply:
Currently the build only works for C++17 compilers. It's not out-of-date.

If you can be more specific in your question I can try to help parse the code.

Reply:
I think it's just my c++ failing me. I'll give it another read.

Reply:
I think one thing that is confusing matters is that if we want an ```fvar<T,0,0,0,0,1>``` we represent the 4 zeros as the size 4 integer sequence [0,1,2,3]. Would it be possible to instead represent it as an integer indicating the number of zeros we want? I'm still wrapping my head around this so the answer could well be no.